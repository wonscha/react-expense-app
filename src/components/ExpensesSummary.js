import { connect } from 'react-redux';
import numeral from 'numeral';
import React from 'react';
import getExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
    </div>
  )
};

const mapStateToProps = (state) => ({
  expenseCount: getVisibleExpenses(state.expenses, state.filters).length,
  expensesTotal: getExpensesTotal(getVisibleExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpensesSummary);