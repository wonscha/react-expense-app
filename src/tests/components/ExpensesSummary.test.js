import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary when one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={9434} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary when multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={9434} />);
  expect(wrapper).toMatchSnapshot();
})