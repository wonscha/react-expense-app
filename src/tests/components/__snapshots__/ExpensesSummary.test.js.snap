// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary when multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $94.34
  </h1>
</div>
`;

exports[`should render ExpensesSummary when one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $94.34
  </h1>
</div>
`;
